### ISSUE TITLE
Build error due to erroneous Eigen types (On commit 2546b89)

### ISSUE THREAD

**Issue by OxDuke** [lmontaut/collision-detection-benchmark#2]
Hi, thanks for open-sourcing the benchmark. After successfully installing all dependencies, I started to build the repo using CMake. The commit is 2546b89. CMake failed and its error message points to [Line 63 in hpp/fcl/include/hpp/fcl/internal/intersect.h](https://github.com/lmontaut/hpp-fcl/blob/340fe6a16b547901c260008824dc65ad6e96ab73/include/hpp/fcl/internal/intersect.h#L63). 
Error gave by the g++ complier was: 
`error: ‘Vector’ in namespace ‘Eigen’ does not name a template type; did you mean ‘VectorXd’?`
My fix was to use `Eigen::Matrix<double, 4, 1>` instead of the `Eigen::Vector<double ,4>`.
My Eigen version is 3.3.7
After the modification, the code complied successfully.

I am writing this issue to make sure my modification is correct.

**Comment by lmontaut** [comment]
Normally you should not have to change the code inside hppfcl.
I tried rerunning everything from a fresh conda env and I can't seem to reproduce this bug.
The version I have for eigen is `3.4.0`.
I successfully built hppfcl both on g++ `12.1.0` and clang++ `13.0.1`.

**Comment by OxDuke** [comment]
I think I found the problem. The `Eigen::Vector<>` template is a new feature of Eigen 3.4.0. See:
https://eigen.tuxfamily.org/index.php?title=3.4#New_Major_Features_in_Core
Here is a screenshot from the website:
![Screenshot from 2022-07-07 09-41-51](https://user-images.githubusercontent.com/19476105/177672176-a533213c-cea8-4c3a-8b2f-f5c530cfd927.png)

So people using Eigen less than 3.4.0 may experience problems during compliation. I also went through the apt website and I found that only Ubuntu22.04 LTS supports Eigen 3.4.0 . See:
https://packages.ubuntu.com/jammy/libeigen3-dev

I wonder if it is fine to use something else for `Eigen::Vector<>` for better compatibility.

**Comment by lmontaut** [comment]
Hi @OxDuke, I am running Manjaro so I just fired up a brand new Ubuntu 20.04 VM to see if I could reproduce your bugs. I installed anaconda and went through the instruction in the readme. When doing `conda install pinocchio`, the right versions of `eigen` and `eigenpy` are found, as expected. For eigen, version `3.4.0` is found by conda, I did not force it.
 Finally, the Ubuntu VM has `g++` version `9.4.0`.
 Hppfcl was built flawlessly. I also did not encounter the boost issue from #3 and was able to run the benchmarks as expected.
